### PR TITLE
Fix singer editor not loading

### LINKED ISSUE #1146: Invest the case that cannot open singer editor
image
here's the log:
```
2022-03-03 13:20:39 [error]: An unhandled error has occurred.
2022-03-03 13:20:39 [error]: osu.Framework.Graphics.Drawable+InvalidThreadForMutationException: Cannot mutate the InternalChildren on a Loaded Drawable while not on the update thread. Consider using Schedule to schedule the mutation operation.
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Drawable.EnsureMutationAllowed(String action)
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.ChangeInternalChildDepth(Drawable child, Single newDepth)
2022-03-03 13:20:39 [error]: at osu.Game.Screens.Edit.Compose.Components.EditorBlueprintContainer.OnBlueprintDeselected(SelectionBlueprint`1 blueprint) in /usr/local/var/buildkite-agent/builds/ppybin-parallels-macos/ppy/osu/osu.Game/Screens/Edit/Compose/Components/EditorBlueprintContainer.cs:line 149
2022-03-03 13:20:39 [error]: at osu.Game.Screens.Edit.Compose.Components.BlueprintContainer`1.<load>b__23_0(Object selectedObjects, NotifyCollectionChangedEventArgs args) in /usr/local/var/buildkite-agent/builds/ppybin-parallels-macos/ppy/osu/osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs:line 58
2022-03-03 13:20:39 [error]: at osu.Framework.Bindables.BindableList`1.removeAll(Predicate`1 match, BindableList`1 caller)
2022-03-03 13:20:39 [error]: at osu.Framework.Bindables.BindableList`1.removeAll(Predicate`1 match, BindableList`1 caller)
2022-03-03 13:20:39 [error]: at osu.Game.Rulesets.Karaoke.Utils.BindablesUtils.<>c__DisplayClass1_0`2.<OnyWaySync>b__0(Object _, NotifyCollectionChangedEventArgs args)
2022-03-03 13:20:39 [error]: at osu.Framework.Bindables.BindableList`1.clear(BindableList`1 caller)
2022-03-03 13:20:39 [error]: at osu.Game.Rulesets.Karaoke.Edit.Singers.SingerScreen.PopOut()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.VisibilityContainer.LoadAsyncComplete()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Drawable.load(IFrameBasedClock clock, IReadOnlyDependencyContainer dependencies)
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Drawable.Load(IFrameBasedClock clock, IReadOnlyDependencyContainer dependencies, Boolean isDirectAsyncContext)
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Drawable.LoadFromAsync(IFrameBasedClock clock, IReadOnlyDependencyContainer dependencies, Boolean isDirectAsyncContext)
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.loadComponents[TLoadable](List`1 components, IReadOnlyDependencyContainer dependencies, Boolean isDirectAsyncContext, CancellationToken cancellation)
2022-03-03 13:20:39 [error]: at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
2022-03-03 13:20:39 [error]: --- End of stack trace from previous location ---
2022-03-03 13:20:39 [error]: at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
2022-03-03 13:20:39 [error]: --- End of stack trace from previous location ---
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.<>c__DisplayClass17_2`1.<LoadComponentsAsync>b__4()
2022-03-03 13:20:39 [error]: at osu.Framework.Threading.ScheduledDelegate.RunTaskInternal()
2022-03-03 13:20:39 [error]: at osu.Framework.Threading.Scheduler.Update()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Drawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
2022-03-03 13:20:39 [error]: at osu.Framework.Platform.GameHost.UpdateFrame()
2022-03-03 13:20:39 [error]: at osu.Framework.Threading.GameThread.processFrame()
2022-03-03 13:20:39 [verbose]: Unhandled exception has been allowed with 0 more allowable exceptions .
2022-03-03 13:20:39 [verbose]: ⚠️ Subsequent messages have been logged. Click to view log files.

```

### PR DESCRIPTION
**Pull request by Joehuu**
- Fixes https://github.com/karaoke-dev/karaoke/issues/1146

Seems to work.

### CODE REVIEW

File: osu.Game.Rulesets.Karaoke/Screens/Edit/Beatmaps/Singers/SingerScreen.cs [hunk 1]
```diff
@@ -40,10 +43,8 @@ public SingerScreen()
     }
 
     [BackgroundDependencyLoader]
-    private void load(EditorBeatmap editorBeatmap, EditorClock editorClock)
+    private void load(EditorClock editorClock)
     {
-        BindablesUtils.Sync(selectedLyrics, editorBeatmap.SelectedHitObjects);
-
```
Comment (andy840119) [inline osu.Game.Rulesets.Karaoke/Screens/Edit/Beatmaps/Singers/SingerScreen.cs:46]: Maybe use `schedule` instead?
e.g.
```
Schedule(() =>
{
    // todo: the reason why use schedule?
    BindablesUtils.Sync(selectedLyrics, editorBeatmap.SelectedHitObjects);
});
```

**Review by andy840119 — COMMENTED** [review]
Thanks for the contribution!
Feel free to make the force-push if needed.

File: osu.Game.Rulesets.Karaoke/Screens/Edit/Beatmaps/Singers/SingerScreen.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game.Rulesets.Karaoke/Screens/Edit/Beatmaps/Singers/SingerScreen.cs:46]: The `InvalidThreadForMutationException` does recommend it, but based on the osu! codebase, bindable logic is generally done in `LoadComplete()` and it looks better and safer IMO.

Using either `Schedule` or `LoadComplete()` ensures it runs in the update thread, but `Schedule` should be used in more direct drawable mutation lines, see https://github.com/ppy/osu-framework/wiki/Threading#scheduling.

Comment (andy840119) [reply osu.Game.Rulesets.Karaoke/Screens/Edit/Beatmaps/Singers/SingerScreen.cs:46]: OK
Seems make sense 👍🏼

**Pull request merged**